sites/docs/src/content/docs/guidelines/pipelines/requirements/semantic_versioning.md

```diff
@@ -6,13 +6,22 @@ menu:
     weight: 90
 ---
 
-Pipelines must be released with stable release tags.
-Releases must use GitHub releases and [keep a detailed changelog](https://keepachangelog.com/en/1.0.0/) file.
+- Pipelines must be released with stable release tags.
+- Releases must use GitHub releases and include a [detailed changelog](https://keepachangelog.com/en/1.0.0/) file.
+- Release version tags must be numerical only (no `v` prefix) and should follow [semantic versioning](https://semver.org/) rules: `[major].[minor].[patch]`
 
-Release version tags must be numerical only (no `v` prefix) and should follow [semantic versioning](https://semver.org/) rules: `[major].[minor].[patch]`
+**Versioning Examples**
 
-For example, starting with with a release version `1.4.3`, bumping the version to:
+Starting from release version `1.4.3`, bumping the version to:
 
-- `1.4.4` would be a patch release for minor things such as fixing bugs.
-- `1.5` would be a minor release, for example adding some new features.
-- `2.0` would correspond to the _major_ release where results would no longer be backwards compatible.
+- **`1.4.4`**: A patch release for minor fixes such as bug corrections that do not modify how the user interacts with the pipeline. Examples:
+  - Inserting a missing argument to a process command line
+  - Updating a container build version of a tool that is otherwise the same version of the tool itself
```

Comment:
What about having a checklist insteaad, something such as:
```suggestion
## Release type checklist

If your release matches any point in the checklists below, classify it according to the highest applicable category in this order: major, minor, then patch. A single match in a higher category overrides all lower categories.

### Major Release (e.g., 1.0.0 to 2.0.0)

- [ ] Changes that are not backwards compatible in terms of pipeline user interaction
- [ ] Renaming or deprecation of existing parameter(s)
- [ ] Changes to input sample sheet specification (adding, dropping, renaming of mandatory columns)
- [ ] Significant changes to the pipeline's core functionality or workflow
- [ ] Major changes to output formats or directory structures
- [ ] Removal of previously supported features or tools
- [ ] Changes that require users to modify their existing configuration files or scripts
- [ ] Upgrades to dependencies that introduce breaking changes

## Minor Release (e.g., 1.4.0 to 1.5.0)

- [ ] Addition of new features without changing existing functionality
- [ ] Adding a new parameter that introduces new functionality/options to an existing module
- [ ] Adding a new optional or default process that doesn't change user interaction with previous steps
- [ ] Introducing new output files or metrics without altering existing ones
- [ ] Adding support for new input formats or data types
- [ ] Implementing performance improvements that don't affect the pipeline's interface
- [ ] Expanding documentation or adding new examples without changing core functionality
- [ ] Adding new optional dependencies or tools

## Patch Release (e.g., 1.4.3 to 1.4.4)

- [ ] Minor fixes such as bug corrections that do not modify how the user interacts with the pipeline
- [ ] Inserting a missing argument to a process command line
- [ ] Updating a container build version of a tool that is otherwise the same version of the tool itself
- [ ] Fixing the logic of pipeline operators to ensure correct inputs for a process
- [ ] Correcting typos in documentation or code comments
- [ ] Updating metadata or non-functional elements (e.g., README, CONTRIBUTING files)
- [ ] Making minor adjustments to log messages or error handling
- [ ] Applying small optimizations that don't affect functionality or user interaction

When deciding on the release type, it should. be considered the impact on users and whether they need to make changes to their existing workflows or configurations. If in doubt, it's generally safer to opt for a higher version increment to clearly communicate the extent of changes to users.
```

Reply:
Yes, that is why I included the message above the checklist. Anyway, my point is more about how having a list should make it easier for developers to see which category the release corresponds to. So a regular list would be also ok.

Reply:
Modified my comment accordingly ☝️

Reply:
ah, didn't read everything, sorry, yes, actually a nice idea

Reply:
I like this idea of a list

Reply:
OK, I'm going to merge this suggestion and we can tweak if necessary from there.